packages/devtools_shared/lib/src/deeplink/deeplink_manager.dart

```diff
@@ -54,4 +64,122 @@ class DeeplinkManager {
       kOutputJsonField: outputJson,
     };
   }
+
+  Future<Map<String, Object?>> getAndroidAppLinkSettings({
+    required String rootPath,
+    required String buildVariant,
+  }) async {
+    final ProcessResult result = await runProcess(
+      'flutter',
+      <String>[
+        'analyze',
+        '--android',
+        '--output-app-link-settings',
+        '--build-variant=$buildVariant',
+        rootPath,
+      ],
+    );
+    if (result.exitCode != 0) {
+      return <String, String>{
+        kErrorField:
+            'Flutter command exit with non-zero error code ${result.exitCode}\n${result.stderr}',
+      };
+    }
+    final match = _outputFilePathRegex.firstMatch(result.stdout);
+    final String outputJson;
+    if (match == null) {
+      return <String, String?>{
+        kErrorField: 'No output file generated: ${result.stderr}',
+      };
+    } else {
+      outputJson = await File(match.group(1)!).readAsString();
+    }
+    try {
+      jsonEncode(outputJson);
+    } on Error catch (e) {
+      return <String, String?>{
+        kErrorField: e.toString(),
+      };
+    }
+    return <String, String?>{
+      kOutputJsonField: outputJson,
+    };
```

Comment:
Most of this is duplicated above and in the methods below. Can we use a helper method to de-dupe?